internal/interpreter/dynamic_component.rs

```diff
@@ -233,11 +245,24 @@ impl<'id> Drop for ErasedComponentBox {
 
 pub type DynamicComponentVRc = vtable::VRc<ComponentVTable, ErasedComponentBox>;
 
+/// This points to the dynamic item tree node that embeds this component.
+#[derive(Clone, Default)]
+pub enum ComponentKind {
+    #[default]
+    Root,
+    SubComponent {
+        parent: ComponentWeak,
+        item_tree_index: usize,
+    },
+}
+
 #[derive(Default)]
 pub(crate) struct ComponentExtraData {
     pub(crate) globals: crate::global_component::GlobalStorage,
     pub(crate) self_weak:
         once_cell::unsync::OnceCell<vtable::VWeak<ComponentVTable, ErasedComponentBox>>,
+    /// This is the global (across all compilation units!) parent_node (if any)
+    pub(crate) parent: core::cell::RefCell<ComponentKind>,
```

Comment:
I thought about this and come to the conclusion that this needs to be a Property<> instead of a RefCell<>. Any binding expression that ends up walking up the tree needs to become dirty when the parent changes.

If it's a property, then it'll also need to be pin-boxed.